centreon/www/class/centreonDowntime.class.php

```diff
@@ -265,211 +265,319 @@ public function getInfos($id)
+                case 'services':
+                    $query = 'SELECT CONCAT(dsr.host_host_id, CONCAT(\'-\', dsr.service_service_id)) AS resource_id,
+                        s.service_activate AS activated
+                    FROM downtime_service_relation dsr
+                    INNER JOIN service s
+                        ON dsr.service_service_id = s.service_id
+                    WHERE
+                        dt_id = :downtimeId';
                     break;
-                case 'svcgrp':
-                    $query = "SELECT sg_sg_id as obj_id FROM downtime_servicegroup_relation WHERE dt_id = :id";
+                case 'servicegroups':
+                    $query = 'SELECT
+                        dsgr.sg_sg_id AS resource_id,
+                        sg.sg_activate AS activated
+                    FROM downtime_servicegroup_relation dsgr
+                    INNER JOIN servicegroup sg
+                        ON dsgr.sg_sg_id = sg.sg_id
+                    WHERE
+                        dt_id = :downtimeId';
                     break;
             }
-            $res = $this->db->prepare($query);
-            $res->bindValue(':id', $id, PDO::PARAM_INT);
-            $res->execute();
-            while ($row = $res->fetch()) {
-                $list[$type][] = $row['obj_id'];
+
+            $statement = $this->db->prepare($query);
+            $statement->bindValue(':downtimeId', $downtimeId, \PDO::PARAM_INT);
+            $statement->setFetchMode(\PDO::FETCH_ASSOC);
+            $statement->execute();
+
+            foreach ($statement as $record) {
+                $relations[$resourceType][] = [
+                    'id' => $record['resource_id'],
+                    'activated' => $record['activated']
+                ];
             }
-            $res->closeCursor();
+
+            $statement->closeCursor();
         }
 
-        return $list;
+        return $relations;
     }
 
-    public function getHostDowntimes()
+    /**
+     * Returns all downtimes configured for enabled hosts
+     *
+     * @return array
+     */
+    public function getForEnabledHosts()
     {
-        $hostDowntimes = array();
-
-        $query = 'SELECT dt.dt_id, dt.dt_activate, dtp.dtp_start_time, dtp.dtp_end_time, dtp.dtp_day_of_week, '
-            . 'dtp.dtp_month_cycle, dtp.dtp_day_of_month, dtp.dtp_fixed, dtp.dtp_duration, '
-            . 'h.host_id, h.host_name, NULL as service_id, NULL as service_description '
-            . 'FROM downtime_period dtp, downtime dt, '
-            . 'downtime_host_relation dtr, host h '
-            . 'WHERE dtp.dt_id = dtr.dt_id AND dtp.dt_id = dt.dt_id '
-            . 'AND dtr.host_host_id = h.host_id';
-
-        try {
-            $res = $this->db->query($query);
-            while ($row = $res->fetch()) {
-                $hostDowntimes[] = $row;
-            }
-        } catch (\PDOException $e) {
-            // Nothing to do
+        $downtimes = [];
+
+        $request = "SELECT dt.dt_id,
```

Comment:
same as previous

Reply:
Okay I'll apply this thanks 💯

Reply:
Suggestion applied thanks for the review 👍